Add links to examples paragraphs

From https://github.com/checkstyle/checkstyle/pull/13112#discussion_r1225682976
> As the discussion in discord, should these PRs add the ID to connect the xdoc and input together?

This is a good idea but I cannot add `id` prop directly to a `source` tag
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.12.1:site (default-site) on project checkstyle: Error parsing '/home/stoyan/open-source/checkstyle/src/xdocs/config_blocks.xml': Error validating the model: Error:
[ERROR]   Public ID: null
[ERROR]   System ID: null
[ERROR]   Line number: 85
[ERROR]   Column number: 26
[ERROR]   Message: cvc-complex-type.3.2.2: Attribute 'id' is not allowed to appear in element 'source'.
```

An option here is to create an invisible div above the example.
```diff
+        <div id="Example1.config" style="display:none;"></div>
        <source>
&lt;module name=&quot;AvoidNestedBlocks&quot;/&gt;
        </source>
```

In addition to this, I see there is a task at https://github.com/orgs/checkstyle/projects/5 to investigate the ability to set links for any place we think. 

Why don't we combine those two?

--- a/src/site/resources/css/site.css
+++ b/src/site/resources/css/site.css
@@ -42,7 +42,7 @@ h4 {
   float: right;
 }
 
-h2:hover .anchor, h3:hover .anchor {
+h2:hover .anchor, h3:hover .anchor, p[id^="Example"]:hover .anchor {
   visibility: visible;
 }
 

--- a/src/site/resources/js/anchors.js
+++ b/src/site/resources/js/anchors.js
@@ -56,6 +56,25 @@
             anchor.appendChild(a);
             anchorItem.appendChild(anchor);
         });
+
+        const exampleDivs = document.querySelectorAll('p[id^="Example"]');
+        [].forEach.call(exampleDivs, function (exampleDiv) {
+            const name = exampleDiv.id;
+            const link = "" + url + "#" + name + "";
+
+            const a = document.createElement("a");
+            a.setAttribute("href", link);
+
+            const image = document.createElement("img");
+            image.setAttribute("src", `${relativePath}/images/anchor.png`);
+
+            const anchor = document.createElement("div");
+            anchor.className = "anchor";
+
+            a.appendChild(image);
+            anchor.appendChild(a);
+            exampleDiv.appendChild(anchor);
+        });
     });
 }());
 

